src/antares/service/local_services/area_local.py

```diff
@@ -334,3 +335,33 @@ def read_areas(self) -> List[Area]:
                     )
                 )
         return areas
+
+    def _read_timeseries(self, ts_file_type: TimeSeriesFileType, study_path: Path, area_id: Optional[str] = None, group_id: Optional[str] = None,constraint_id: Optional[str] = None, time_series: Optional[pd.DataFrame] = None,) -> pd.DataFrame:
+        def _write_file(_file_path: Path, _time_series: pd.DataFrame) -> None:
+            _file_path.parent.mkdir(parents=True, exist_ok=True)
+            _time_series.to_csv(_file_path, sep="\t", header=False, index=False, encoding="utf-8")
+        
+        file_path = study_path / (
+            ts_file_type.value
+            if not (area_id or constraint_id or group_id)
+            else ts_file_type.value.format(area_id=area_id, constraint_id=constraint_id, group_id=group_id)
+        )
+        if file_path.is_file() and time_series is None:
+            if os.path.getsize(file_path) != 0:
+                _time_series = pd.read_csv(
+                    file_path,
+                    sep="\t",
+                    header=None,
+                    index_col=None,
+                    encoding="utf-8",
+                )
+            else:
+                _time_series = pd.DataFrame()
+        else:
+            _time_series = time_series if time_series is not None else pd.DataFrame([])
+            _write_file(file_path, _time_series)
```

Comment:
Why would you want to write a file inside a read method ? You shouldn't do this.

Also you could use the `df_read` method introduced recently to centralize the code relative to dataframes.
You could add the if else with the file_size inside this method  as it's useful

Reply:
It is to replace the class TimeSeriesFile which did this